### PR TITLE
Document conversions between Rust and SQL Server

### PR DESCRIPTION
**Pull request by pimeys**
Should address the issue https://github.com/prisma/tiberius/issues/59

### CODE REVIEW

File: src/to_sql.rs [hunk 1]
```diff
@@ -6,6 +6,44 @@ use std::borrow::Cow;
...
+/// |`String`/`&str` (< 4000 characters)|`nvarchar(4000)`|
+/// |`String`/`&str`|`nvarchar(max)`|
+/// |`Uuid`|`uniqueidentifier`|
+/// |`Vec<u8>`/`&[u8]` (< 8000 bytes)|`varbinary(8000)`|
+/// |`Vec<u8>`/`&[u8]`|`varbinary(max)`|
+/// |`Numeric`|`numeric`/`decimal`|
+/// |`Decimal`|`numeric`/`decimal`|
+/// |`XmlData`|`xml`|
+/// |`NaiveDate` (TDS 7.3 >)|`date`|
+/// |`NaiveTime` (TDS 7.3 >)|`time`|
```
Comment (yoshuawuyts) [inline src/to_sql.rs:32]: What does TDS mean here?

File: src/from_sql.rs [hunk 2]
```diff
@@ -2,6 +2,33 @@ use crate::{tds::Numeric, xml::XmlData, ColumnData};
...
+/// |`bool`|`bit`|
+/// |`String`/`&str`|`nvarchar`/`varchar`/`nchar`/`char`/`ntext`/`text`|
+/// |`Uuid`|`uniqueidentifier`|
+/// |`Vec<u8>`/`&[u8]`|`binary`/`varbinary`/`image`|
+/// |`Numeric`|`numeric`/`decimal`|
+/// |`Decimal`|`numeric`/`decimal`|
+/// |`XmlData`|`xml`|
+/// |`NaiveDateTime`|`datetime`/`datetime2`/`smalldatetime`|
+/// |`NaiveDate`|`date`|
+/// |`NaiveTime`|`time`|
```
Comment (yoshuawuyts) [inline src/from_sql.rs:27]: Where does `NaiveDateTime` come from? Do we re-export these?

File: src/to_sql.rs (hunk 1 above)
Comment (pimeys) [reply src/to_sql.rs:32]: Microsoft slang. TDS (Tabular Data Stream) is the protocol, and a bit underneath it's explained `SQL Server 2005` uses the version 7.2, later versions are TDS 7.3 or 7.4. The new date and time types were introduced in 7.3.

File: src/from_sql.rs [hunk 3]
```diff
@@ -2,6 +2,33 @@ use crate::{tds::Numeric, xml::XmlData, ColumnData};
 use uuid::Uuid;
 
 /// A conversion trait from a TDS type by-reference.
+///
+/// An implementation to a Rust type is needed for using it as a return
```
Comment (tomhoule) [inline src/from_sql.rs:6]: absolute nitpick: maybe `a FromSql implementation for a Rust type is needed`

File: src/from_sql.rs (hunk 2 above)
Comment (pimeys) [reply src/from_sql.rs:27]: This is the question. The [time module](https://docs.rs/tiberius/0.4.4/tiberius/time/index.html) explains that one can in theory use the provided date and time types, but they are quite useless and the preferred way is to use the `chrono` feature flag with `chrono` types if needing to manipulate any of these values.

Comment (pimeys) [reply src/from_sql.rs:27]: Should we re-export these btw?

Comment (yoshuawuyts) [reply src/from_sql.rs:27]: yeah, I think so -- they're a peer dependency already, so may as well make it easier to use (and ensure people don't run into version mismatches).

**Pull request merged**